ddl/ddl.go

```diff
@@ -394,6 +394,12 @@ func (sv *schemaVersionManager) setSchemaVersion(job *model.Job, store kv.Storag
 		var err error
 		m := meta.NewMeta(txn)
 		schemaVersion, err = m.GenSchemaVersion()
+		if err != nil {
+			return err
+		}
+
+		logutil.BgLogger().Info("DDL create schema with timestamp", zap.Int64("version", schemaVersion), zap.Uint64("start", txn.StartTS()))
+		err = m.SetTimestampForSchemaVersion(int64(txn.StartTS()), schemaVersion)
```

Comment:
I am afraid that you need `commitTS`.  Check `GetSchemaVersionWithNonEmptyDiff`. You need a timestamp where schemadiff is actually set.

This is a sub txn updating schema version only. That means the actual schemadiff and metadata are not writtent in this txn. I think the correct txn is at https://github.com/pingcap/tidb/blob/541ccd5013d2303a4c3e0a1793c6f16b73c25e30/ddl/ddl_worker.go#L813, where `w.sess.commit` is successfully invoked.

There is one RPC avaiable for that. Check https://github.com/pingcap/tidb/blob/d6530113280f5b019814642b7587689e90fd8bf5/docs/tidb_http_api.md, `mvcc/txn/{startTS}/{db}/{table}`.  Also this:

https://github.com/pingcap/tidb/blob/d6530113280f5b019814642b7587689e90fd8bf5/store/helper/helper.go#L127